langlib/lang.array/src/main/ballerina/array.bal

```diff
@@ -46,8 +50,15 @@ public isolated function length((any|error)[] arr) returns int = @java:Method {
 
 # Returns an iterator over an array.
 #
+# ```
+# object {
+#     public isolated function next() returns record {|int value;|}?;
+# } iterator = [2, 4, 6, 8].iterator();
+# iterator.next() => {"value":2}
```

Comment:
Had to call `next` to get the string representation of something meaningful. Same for `toStream`.